### ISSUE TITLE
question: does Refit support C# 9 record?

### ISSUE THREAD

**Issue by roubachof** [reactiveui/refit#1208]
C#9 records are finally the immutable type that we needed in C#.
Since it's syntax is game changing, does refit supports serializing/deserializing such objects ?

**Comment by laurentksh** [comment]
I think they're already supported as, from my understanding, records are classes under the hood.

**Comment by vmachacek** [comment]
Does it work with [AliasAs] attribute? I had some issued with it when used like this:

```
public record AuthRequest([AliasAs("grant_type")] string GrantType,
    [AliasAs("scope")] string Scope,
    [AliasAs("client_id")] string ClientId,
    [AliasAs("client_secret")] string ClientSecret);
```

In this case the [AliasAs] is not reflected and the content looks like this

`GrantType=client_credentials&Scope=read%3Aall&ClientId=foobar&ClientSecret=foobar`

but should look like 

`grant_type=client_credentials&scope=read%3Aall&client_id=foobar&client_secret=foobar`

**Comment by starushykart** [comment]
Hi! 
I have the same issue. [AliasAs] does not work with records and their primary c-tors

**Comment by djekmusic** [comment]
JsonProperty doesn't work with records also

**Comment by glennawatson** [comment]
Yes - records work fine as request and response models; they're classes under the hood and serialize/deserialize normally.

One caveat raised in this thread: putting `[AliasAs("...")]` directly on a record's primary-constructor parameter does not rename form/query fields, because the attribute lands on the constructor parameter, not the generated property. Use the `property:` target so the attribute is emitted onto the property:

    public record AuthRequest([property: AliasAs("grant_type")] string GrantType);

That is standard C# attribute-target behavior, not a Refit limitation. (For JSON body deserialization, AliasAs is not the right tool anyway - use the serializer's naming attribute, e.g. `JsonPropertyName`/`JsonProperty`.)

Closing the original question as answered. If you want AliasAs to auto-flow from primary-ctor parameters without `[property:]`, that's a separate enhancement - please open a focused issue or Discussions thread: https://github.com/reactiveui/refit/discussions


<blockquote><img src="https://opengraph.githubassets.com/7da521d756b94831ff31d8d5fc75ecbe2c2d69a69928da79595847192e775278/reactiveui/refit" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/reactiveui/refit/discussions">reactiveui refit · Discussions</a></strong></div><div>Explore the GitHub Discussions forum for reactiveui refit. Discuss code, ask questions & collaborate with the developer community.</div></blockquote>
